[OldMongo FC-0004] Remove support for children in Old Mongo

UvgenGen opened the pull request:
This commit leaves behind just enough Old Mongo (DraftModulestore)
functionality to allow read-only access to static assets and the
root CourseBlock. It removes:

* create/update operations
* child/parent traversal
* inheritance related code

It also removes or converts tests for this functionality.

The ability to read from the root CourseBlock was maintained for
backwards compatibility, since top-level course settings are often
stored here, and this is used by various parts of the codebase,
like displaying dashboards and re-building CourseOverview models.

Any attempt to read the contents of a course by getting the
CourseBlock's children will return an empty list (i.e. it will look
empty).

This commit does _not_ delete content on MongoDB or run any sort of
data migration or cleanup.

----

Useful information to include:
https://github.com/openedx/public-engineering/issues/80

openedx-webhooks commented:
Thanks for the pull request, @UvgenGen!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

ormsbee commented on common/djangoapps/student/tests/test_certificates.py:
```diff
@@ -76,6 +76,7 @@ def _create_certificate(self, enrollment_mode):
         )
 
 
+@unittest.skip("OldMongo Deprecation")
```
Shouldn't the certificates tests still run the same as before? Does this PR break certs functionality for Old Mongo somehow?

UvgenGen replied:
Problem in `test_certificate_available_date` test in case where `past_date` is `False`.
Looks like [_check_message](https://github.com/openedx/edx-platform/pull/31134/files#diff-2d7a93c69b7e26711a41032845bd2b186f0e5a651037c0693cd308bb5033617aR96) checker not correct, this checker expect one message ('Your grade and certificate will be ready after') for both cases but message with `is_past` flag `False` should be 'Congratulations! Your certificate is ready.'
I'll fix it

ormsbee commented on common/djangoapps/student/tests/test_course_listing.py:
```diff
@@ -105,12 +105,13 @@ def test_errored_course_regular_access(self):
-        with mock.patch('xmodule.modulestore.mongo.base.MongoKeyValueStore', mock.Mock(side_effect=Exception)):
+        with mock.patch(
+            'xmodule.modulestore.split_mongo.caching_descriptor_system.SplitMongoKVS', mock.Mock(side_effect=Exception)
+        ):
```
We should also keep testing that course listings don't break with Old Mongo courses (they'll still appear on the Learner Dashboard, even if they're not clickable).

UvgenGen replied:
We can't create the old mongo test course because `update_item` is deprecated and raised NotImplementedError

ormsbee commented on lms/djangoapps/course_blocks/transformers/tests/test_load_override_data.py:
```diff
@@ -52,9 +52,6 @@ def setUp(self):
         CourseEnrollmentFactory.create(user=self.learner, course_id=self.course_key, is_active=True)
-        self.block = self.store.create_child(
-            self.learner2.id, subsection.location, 'html', 'new_component'
-        )
```
Why was it necessary to delete this?

UvgenGen replied:
returned with changed name

ormsbee commented on cms/djangoapps/contentstore/management/commands/tests/test_cleanup_assets.py:
```diff
@@ -20,6 +21,7 @@
 TEST_DATA_DIR = settings.COMMON_TEST_DATA_ROOT
 
 
+@skip("OldMongo Deprecation")
```
Is this marked for skipping because it's a test that currently only works for Old Mongo but may later be converted to work with Split?

ormsbee commented on cms/djangoapps/contentstore/management/commands/tests/test_create_course.py:
```diff
@@ -46,21 +46,23 @@ class TestCreateCourse(ModuleStoreTestCase):
     Unit tests for creating a course in either old mongo or split mongo via command line
     """
 
-    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
```
Since you're removing the `@ddt.data` decorator here, there's no need for the class itself to have a `@ddt.ddt` decorator.

UvgenGen replied:
updated

ormsbee commented on cms/djangoapps/contentstore/management/commands/tests/test_create_course.py:
```diff
@@ -46,21 +46,23 @@ class TestCreateCourse(ModuleStoreTestCase):
         self.assertTrue(
             modulestore().has_course(new_key),
-            f"Could not find course in {store}"
+            f"Could not find course in {ModuleStoreEnum.Type.split,}"
```
Nit: Why the comma at the end?

UvgenGen replied:
updated

ormsbee commented on cms/djangoapps/contentstore/management/commands/tests/test_delete_orphans.py:
```diff
@@ -23,26 +23,24 @@ def test_no_args(self):
         with self.assertRaisesRegex(CommandError, errstring):
             call_command('delete_orphans')
 
-    @ddt.data(ModuleStoreEnum.Type.split, ModuleStoreEnum.Type.mongo)
```
Please also remove the `@ddt.ddt` decorator on the class.

UvgenGen replied:
updated in https://github.com/openedx/edx-platform/pull/31224

ormsbee commented on cms/djangoapps/contentstore/management/commands/tests/test_export.py:
```diff
@@ -45,16 +45,15 @@ def setUp(self):
 
-    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
-    def test_export_course_with_directory_name(self, store):
+    def test_export_course_with_directory_name(self):
```
Please also remove the `@ddt.ddt` decorator on the class.

UvgenGen replied:
updated in https://github.com/openedx/edx-platform/pull/31224

ormsbee commented on cms/djangoapps/contentstore/management/commands/tests/test_export_all_courses.py:
```diff
@@ -13,6 +14,7 @@
 from xmodule.modulestore.tests.factories import CourseFactory  # lint-amnesty, pylint: disable=wrong-import-order
 
 
+@skip("OldMongo Deprecation")
```
Should we just delete this test entirely? What breaks when trying to convert to Split?

UvgenGen replied:
I've got error  for the [line 46 ](https://github.com/openedx/edx-platform/pull/31134/files#diff-84d88e4add49f47a27fabb80407a1439ba5c75cc97222908816049b01302fff8R46)
`AttributeError: 'MixedModuleStore' object has no attribute 'collection'`
split module store has no `collection` attribute.

ormsbee commented on cms/djangoapps/contentstore/management/commands/tests/test_export_olx.py:
```diff
@@ -91,9 +90,8 @@ def test_export_course(self, store_type):
-    def test_export_course_stdout(self, store_type):
-        test_course_key = self.create_dummy_course(store_type)
+    def test_export_course_stdout(self):
+        test_course_key = self.create_dummy_course(ModuleStoreEnum.Type.split)
```
Please also remove the `@ddt.ddt` decorator on the class.

UvgenGen replied:
updated in https://github.com/openedx/edx-platform/pull/31224

ormsbee commented on cms/djangoapps/contentstore/management/commands/tests/test_git_export.py:
```diff
@@ -27,6 +27,7 @@
 TEST_DATA_CONTENTSTORE['DOC_STORE_CONFIG']['db'] = 'test_xcontent_%s' % uuid4().hex
 
 
+@unittest.skip("OldMongo Deprecation")
```
Does this specific test not work with Split, or is `git_export` broken entirely for split mongo courses?

UvgenGen replied:
[test_git_ident](https://github.com/openedx/edx-platform/pull/31134/files#diff-bd41b17cc0061ebc2b45d7f30decbd2312164f0d3aa7f3712426d33bbf81e338R160) test not work due to [self.populate_course()](https://github.com/openedx/edx-platform/pull/31134/files#diff-bd41b17cc0061ebc2b45d7f30decbd2312164f0d3aa7f3712426d33bbf81e338R160). It's related to problem with `get_children()`(return empty list) in tests for split modulestore.

ormsbee commented on cms/djangoapps/contentstore/tests/test_contentstore.py:
```diff
@@ -212,7 +213,7 @@ def test_asset_import(self):
         self.assertGreater(len(all_thumbnails), 0)
 
-        location = AssetKey.from_string('/c4x/edX/toy/asset/just_a_test.jpg')
+        location = AssetKey.from_string('asset-v1:edX+toy+2012_Fall+type@asset+block@just_a_test.jpg')
```
We should continue to test that Old Mongo assets work until we actually remove that functionality (after extracting the certificates data).

UvgenGen replied:
I tried it but we can't use `import_course_from_xml` because `update_item` not implemented now for oldmongo modulestore.

ormsbee commented on cms/djangoapps/contentstore/tests/test_contentstore.py:
```diff
@@ -302,6 +304,7 @@ def verify_content_existence(self, store, root_dir, course_id, dirname, category
             filesystem = OSFS(root_dir / ('test_export/' + dirname))
             self.assertTrue(filesystem.exists(item.location.block_id + filename_suffix))
 
+    @skip("OldMongo Deprecation")
```
Is there a Split equivalent of this test (roundtrip export/import) somewhere else? If so, can we delete this?

UvgenGen replied:
Yes we have `xmodule/modulestore/tests/test_cross_modulestore_import_export.py::CrossStoreXMLRoundtrip`, so I think we can remove this test.

ormsbee commented on cms/djangoapps/contentstore/tests/test_course_create_rerun.py:
```diff
@@ -114,67 +112,61 @@ def test_rerun(self):
 
-    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
-    def test_newly_created_course_has_web_certs_enabled(self, store):
+    def test_newly_created_course_has_web_certs_enabled(self):
```
Please remove `ddt` decorator from class if it's not being used.

UvgenGen replied:
It's used in [line 171](https://github.com/openedx/edx-platform/pull/31134/files/036a49afcf816ba4fb0534bfc8078e9b0959e22c#diff-9b7db26033d96241e179605755d725ac0d2cd6d9b1f29fded65e1299b3a377b7R171)

ormsbee commented on cms/djangoapps/contentstore/tests/test_exams.py:
```diff
@@ -55,6 +56,7 @@ def setUp(self):
         return f'{settings.EXAMS_SERVICE_URL}/exams/course_id/{course_id}/'
 
+    @skip("OldMongo Deprecation")
     @ddt.data(
```
@UvgenGen: We shouldn't skip this test. This is new, actively developed functionality that should be tested via the new Modulestore.

@Zacharis278: Could you please switch this test to `TEST_DATA_SPLIT_MODULESTORE` on master and then we can rebase this PR branch?

zacharis278 replied:
ok so I found what's going on and put up a PR to fix these tests specifically: https://github.com/openedx/edx-platform/pull/31223

Tests related to proctoring scaffold out test data using the field `is_proctored_exam` which is an alias to the underlying `is_proctored_enabled` value. This alias appears to not work when setting up data in split modulestore. It's not entirely obvious to me if this intended, a bug, or this one field is a really gross edge case. The fastest fix is to just use the underlying field name instead so I'm just doing that in my linked PR. This is likely the cause of a few other skipped tests in here.

I've also left a comment [here](https://github.com/openedx/edx-platform/pull/31134/files#r1007355913) because this hiding some tests that should be failing

UvgenGen replied:
updated

ormsbee commented on cms/djangoapps/contentstore/tests/test_export_git.py:
```diff
@@ -97,6 +98,7 @@ def test_exception_translation(self):
         response = self.client.get(f'{self.test_url}?action=push')
         self.assertNotContains(response, 'django.utils.functional.__proxy__')
 
+    @skip("OldMongo Deprecation")
```
I'm a little curious why this failure is modulestore-specific. 😞

UvgenGen replied:
Problem with `get_children()` for split modulestore.

ormsbee commented on cms/djangoapps/contentstore/tests/test_import_pure_xblock.py:
```diff
@@ -66,22 +55,12 @@ def _assert_import(self, course_dir, expected_field_val, has_draft=False):
 
         xblock_location = courses[0].id.make_usage_key('stubxblock', 'xblock_test')
 
-        if has_draft:
```
Why is the `if has_draft` conditional logic removed here?

UvgenGen replied:
According to the [line 67](https://github.com/openedx/edx-platform/pull/31134/files#diff-267f37302ba31751083dbbe407149a2a405e79940a9d8b0a22d9cf01d650dcb9L67) split doesn't work with the "has_draft" logic. Only "old mongo" modulestore has `has_draft` method.

ormsbee commented on cms/djangoapps/contentstore/tests/test_orphan.py:
```diff
@@ -104,7 +103,6 @@ def test_get_orphans(self, default_store):
         (ModuleStoreEnum.Type.split, 5, 3),
-        (ModuleStoreEnum.Type.mongo, 34, 11),
     )
     @ddt.unpack
```
Nit: There's no real need to keep this in ddt if there's only one set of values.

UvgenGen replied:
updated in https://github.com/openedx/edx-platform/pull/31224

ormsbee commented on cms/djangoapps/contentstore/tests/test_users_default_role.py:
```diff
@@ -92,6 +94,7 @@ def test_user_role_on_course_recreate(self):
         # check that user has his default "Student" forum role for this course
         self.assertTrue(self.user.roles.filter(name="Student", course_id=self.course_key))
 
+    @skip("OldMongo Deprecation")
```
😞 We're probably going to want to come back around to this one later, but I agree on skipping if it doesn't work now.

ormsbee commented on cms/djangoapps/contentstore/tests/test_utils.py:
```diff
@@ -411,6 +411,7 @@ def verify_all_components_visible_to_all():
 
         verify_all_components_visible_to_all()
 
+    @skip("OldMongo Deprecation")
```
Why does this need to get skipped? Wasn't this test already running as Split?

UvgenGen replied:
Problem with parent/children relation for split modulestore tests.

ormsbee commented on cms/djangoapps/contentstore/tests/utils.py:
```diff
@@ -157,17 +164,13 @@ def import_and_populate_course(self):
         # create a Published (no draft) vertical
         public_vertical = self.store.create_item(self.user.id, course_id, 'vertical', self.PUBLISHED_VERTICAL)
         public_vertical = self.store.publish(public_vertical.location, self.user.id)
-        self.assertTrue(self.store.has_published_version(public_vertical))
```
Why was it necessary to remove the assertions here?

UvgenGen replied:
`import_and_populate_course` is only used in `test_export_course_roundtrip`. If we remove that test, I think we can remove this method as well.

ormsbee commented on cms/djangoapps/contentstore/views/tests/test_assets.py:
```diff
@@ -73,6 +73,7 @@ def get_sample_asset(self, name, asset_type='text'):
         return sample_asset
 
 
+@skip("OldMongo Deprecation")
```
We need to keep running the basic tests I think, to continue to support the cert image use case.

UvgenGen replied:
I've got NotImplementedError. `update_item` not implemented for oldmongo modulestore now.

ormsbee commented on cms/djangoapps/contentstore/views/tests/test_container_page.py:
```diff
@@ -190,19 +191,20 @@ def _create_item(self, parent_location, category, display_name, **kwargs):
             **kwargs
         )
 
+    @skip("OldMongo Deprecation")
```
Wasn't this test already running as Split? Why is this skip necessary?

UvgenGen replied:
Previously used oldmongo modulestore. Problem with parent/children relation for split modulestore tests.

ormsbee replied:
For anything skipped, please put a short comment in the code explaining why, for the sake of future maintainers who might be looking at fixing/removing later.

ormsbee commented on cms/djangoapps/contentstore/views/tests/test_course_index.py:
```diff
@@ -96,6 +96,7 @@ def _assert_library_tab_present(response):
         index_response = self.client.get(index_url, {}, HTTP_ACCEPT='text/html')
         _assert_library_tab_present(index_response)
 
+    @skip("OldMongo Deprecation")
```
Why do the staff access check tests fail? Are there other staff access checks that do work with split, or would we be skipping all Studio user access check testing by doing this?

UvgenGen replied:
updated

ormsbee commented on cms/djangoapps/contentstore/views/tests/test_course_index.py:
```diff
@@ -945,6 +951,7 @@ def test_indexing_seq_error_responses(self, mock_index_dictionary):
             CoursewareSearchIndexer.do_course_reindex(modulestore(), self.course.id)
 
+    @skip("OldMongo Deprecation")
     @mock.patch('xmodule.modulestore.mongo.base.MongoModuleStore.get_course')
```
Does this work with the Split Mongo equivalent patched?

UvgenGen replied:
It's work, updated

ormsbee commented on cms/djangoapps/contentstore/views/tests/test_group_configurations.py:
```diff
@@ -770,37 +771,29 @@ def test_can_get_correct_usage_info_for_content_groups(self):
         # Assert that actual content group information is same as expected one.
         self.assertEqual(actual, expected)
 
+    @skip("OldMongo Deprecation")
```
Is it just the URLs that break for this one under Split?

UvgenGen replied:
The problem was in the order of the groups. updated

ormsbee commented on cms/djangoapps/contentstore/views/tests/test_import_export.py:
```diff
@@ -749,36 +749,11 @@ def test_unknown_xblock_top_level(self):
-            '/not_a_real_block_type/' in path
-            for path in tar_ball.getnames()
-        )
+        with self.assertRaisesMessage(PluginMissingError, 'not_a_real_block_type'):
```
This seems like a very big change in behavior. The intention was that unknown XBlocks should not break the import/export process, so that people could still mostly use their content between instances or as XBlocks are installed/uninstalled over time. This looks like it breaks that functionality?

ormsbee replied:
Going back to this: is this another case where it was already broken for Split? Did exports with unknown types always trigger PluginMissingErrors? (If so, that's a problem, but it's one we can address elsewhere.) But if this is behavior that's changing because of any recent work, that's a problem.

UvgenGen replied:
PluginMissingError raised in [partition_fields_by_scope](https://github.com/raccoongang/edx-platform/blob/sagirov/EDXOLDMNG-146/xmodule/modulestore/__init__.py#L1274) when we don't provide default class for `XBlock.load_class`. This change was added in this [PR](https://github.com/openedx/edx-platform/pull/2738/files#diff-cf8818b408e493ee91ad33812cd69f6dd8ae788cfb3dac5088cead1e9e79843aR579).

UvgenGen replied:
I added a commit with fix, please take a look)

ormsbee commented on cms/djangoapps/contentstore/views/tests/test_item.py:
```diff
@@ -92,13 +93,11 @@ def setUp(self):
         """
         item = self.store.get_item(usage_key)
-        if verify_is_draft:
-            self.assertTrue(getattr(item, 'is_draft', False))
```
Is the `is_draft` attribute specific to Old Mongo? Split has a notion of drafts as well, doesn't it?

ormsbee replied:
In general, this module seems to test a lot of draft vs. published behavior, and it's not clear to me why that breaks under split?

ormsbee commented on cms/djangoapps/contentstore/views/tests/test_preview.py:
```diff
@@ -312,6 +307,7 @@ def test_anonymous_user_id_individual_per_student(self):
         )
         assert runtime.anonymous_student_id == '26262401c528d7c4a6bbeabe0455ec46'
 
+    @skip("OldMongo Deprecation")
```
How does this one break?

UvgenGen replied:
[Problem with ItemFactory](https://github.com/openedx/edx-platform/pull/31134/files#diff-6ffe337ebfbf7cca4b05ec0982dfc8f5ae8b463b740416d6857e08154cb66751R317) `AttributeError: type object 'LTIBlockWithMixins' has no attribute 'spec'`

ormsbee commented on cms/lib/xblock/test/test_authoring_mixin.py:
```diff
@@ -178,6 +180,7 @@ def test_html_false_content_group(self):
             [self.STAFF_LOCKED]
         )
 
+    @skip("OldMongo Deprecation")
```
What's the exception thrown when this runs under split?

UvgenGen replied:
Looks like only `self.STAFF_LOCKED` string not exist in html.
```
cms/lib/xblock/test/test_authoring_mixin.py:114: in verify_visibility_view_contains
    self.assertIn(string, html)
E   AssertionError: 'The unit that contains this component is hidden from learners' not found in '\n\n<div class="modal-section visibility-summary">\n</div>\n\n    <form class="visibility-controls-form" method="post" action="">\n        <div class="modal-section visibility-controls">\n            <h3 class="modal-section-title visibility-header" id="visibility-title">\n                <span class="current-visibility-title">\n                    <span class="icon fa fa-eye" aria-hidden="true"></span>\n                    <span>Access is restricted to:</span>\n                </span>\n                <span>Deleted Group</span>\n            </h3>\n            <div class="modal-section-content partition-visibility">\n                <label class="group-select-title">Restrict access to:\n                    <select>\n                        <option value="-1" selected ="selected">\n                                All Learners and Staff\n                        </option>\n                            <option value="100" id="visibility-partition-100" selected=&#34;selected&#34;}>\n                                Content Groups\n                            </option>\n                    </select>\n                </label>\n\n                <div role="group" aria-labelledby="partition-group-directions-100" aria-describedby="visibility-partition-100"\n                    class="partition-group-control partition-group-control-100 ">\n                    <div class="partition-group-directions" id="partition-group-directions-100">Select one or more groups:\n                        <div class="field partition-group-visibility partition-group-visibility-100 ">\n                            <input type="checkbox"\n                                id="visibility-group-100-1"\n                                name="visibility-group"\n                                value="1"\n                                class="input input-checkbox"\n                                \n                            />\n                            <label for="visibility-group-100-1" class="label">Cat Lovers</label>\n                         </div>\n                        <div class="field partition-group-visibility partition-group-visibility-100 ">\n                            <input type="checkbox"\n                                id="visibility-group-100-2"\n                                name="visibility-group"\n                                value="2"\n                                class="input input-checkbox"\n                                \n                            />\n                            <label for="visibility-group-100-2" class="label">Dog Lovers</label>\n                         </div>\n                        <div class="field partition-group-visibility partition-group-visibility-100 was-removed">\n                            <input type="checkbox"\n                                id="visibility-group-100-false_group_id"\n                                name="visibility-group"\n                                value="false_group_id"\n                                class="input input-checkbox"\n                                checked=&#34;checked&#34;\n                            />\n                            <label for="visibility-group-100-false_group_id" class="label">\n                                Deleted Group\n                                <span class="note">\n                                    This group no longer exists.  Choose another group or remove the access restriction.\n                                </span>\n                            </label>\n                         </div>\n                    </div>\n                </div>\n            </div>\n        </div>\n    </form>\n'
```

ormsbee replied:
Oh, we changed the error message and never updated this. That's pretty fragile. Just delete the test altogether please.

ormsbee commented on lms/djangoapps/course_api/tests/test_serializers.py:
```diff
@@ -40,45 +40,47 @@ def setUp(self):
-                },
-            },
+            'short_description': None,
+            'media': OrderedDict([
```
I'm curious why OrderedDict was necessary here. Aren't regular Python dicts already in insertion-order since Python 3.7?

UvgenGen replied:
Updated

ormsbee commented on lms/djangoapps/course_wiki/tests/test_access.py:
```diff
@@ -18,22 +18,22 @@
         self.wiki = get_or_create_root()
 
-        self.course_math101 = CourseFactory.create(org='org', number='math101', display_name='Course', metadata={'use_unique_wiki_id': 'false'})  # lint-amnesty, pylint: disable=line-too-long
+        self.course_math101 = CourseFactory.create(org='org', number='math101', display_name='Course')  # lint-amnesty, pylint: disable=line-too-long
```
Why was this change made?

UvgenGen replied:
I was getting an error `AttributeError: type object 'CourseBlockWithMixins' has no attribute 'use_unique_wiki_id'`

ormsbee commented on lms/djangoapps/course_wiki/tests/test_middleware.py:
```diff
@@ -14,15 +14,15 @@
 
         self.wiki = get_or_create_root()
 
-        self.course_math101 = CourseFactory.create(org='edx', number='math101', display_name='2014', metadata={'use_unique_wiki_id': 'false'})  # lint-amnesty, pylint: disable=line-too-long
```
Why was this change made?

UvgenGen replied:
`'CourseBlockWithMixins' has no attribute 'use_unique_wiki_id'`

ormsbee commented on lms/djangoapps/courseware/management/commands/tests/test_dump_course.py:
```diff
@@ -223,15 +222,6 @@ def check_export_file(self, tar_file):  # pylint: disable=missing-function-docst
-    MODULESTORE = TEST_DATA_MONGO_MODULESTORE
-    __test__ = True
-
-
```
🎉

ormsbee commented on lms/djangoapps/courseware/tests/helpers.py:
```diff
@@ -99,7 +99,7 @@ def initialize_module(self, runtime_kwargs=None, **kwargs):  # lint-amnesty, pyl
         self.item_url = str(self.item_descriptor.location)
 
     def setup_course(self):  # lint-amnesty, pylint: disable=missing-function-docstring
-        self.course = CourseFactory.create(data=self.COURSE_DATA)
```
Does removing the `data` arg here mean we're creating a different course structure than before?

UvgenGen replied:
It's the same course structure: COURSE_DATA = {}. COURSE_DATA not used in any other place, so I'll remove this attribute.

ormsbee commented on lms/djangoapps/courseware/tests/test_video_handlers.py:
```diff
@@ -143,6 +143,8 @@ def initialize_block(self, data=None, **kwargs):
             fields_data = VideoBlock.parse_video_xml(data)
             kwargs.update(fields_data)
+        kwargs.pop('source', None)
+        kwargs.get('metadata', {}).pop('source', None)
```
Why were these lines added?

UvgenGen replied:
For example for `TestEditorSavedMethod::test_editor_saved_when_html5_sub_not_exist` test I've got this error:
```
self = <xmodule.modulestore.split_mongo.split_draft.DraftVersioningModuleStore object at 0x7f6dfef87130>
category = 'course', fields = {'data': {}, 'display_name': 'Run 3332'}

    def partition_fields_by_scope(self, category, fields):
        """
        Return dictionary of {scope: {field1: val, ..}..} for the fields of this potential xblock
    
        :param category: the xblock category
        :param fields: the dictionary of {fieldname: value}
        """
        result = defaultdict(dict)
        if fields is None:
            return result
        classes = XBlock.load_class(category, select=prefer_xmodules)
        cls = self.mixologist.mix(classes)
        for field_name, value in fields.items():
>           field = getattr(cls, field_name)
E           AttributeError: type object 'CourseBlockWithMixins' has no attribute 'data'
```
The same for  `source` filed.('property' object has no attribute 'scope'. `cms/djangoapps/contentstore/tests/test_proctoring.py`)
CourseBlockWithMixins just doesn't have these fields or it's not field but property.

ormsbee commented on lms/djangoapps/courseware/tests/test_views.py:
```diff
@@ -167,7 +166,7 @@ def test_jump_to_invalid_location(self, preview_mode, store_type):
 
     @set_preview_mode(True)
-    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    @ddt.data(ModuleStoreEnum.Type.split)
```
Nit: don't need `@ddt.data` for one arg.

UvgenGen replied:
updated in https://github.com/openedx/edx-platform/pull/31224

ormsbee commented on lms/djangoapps/courseware/tests/tests.py:
```diff
@@ -131,7 +131,7 @@ def test_toy_textbooks_loads(self, mock_get):
             </table_of_contents>
         """).strip()
-        location = self.toy_course_key.make_usage_key('course', '2012_Fall')
+        location = self.toy_course_key.make_usage_key('course', 'course')
```
Just checking: Is this because the derivation of the CourseBlock's UsageKey changed between Old Mongo and Split to this convention?

ormsbee commented on lms/djangoapps/discussion/tests/test_views.py:
```diff
@@ -1923,11 +1924,12 @@ def create_divided_discussions(self):
         return divided_inline_discussions, divided_course_wide_discussions
 
 
+@skip("OldMongo Deprecation")
```
This whole class breaks? 😞 May I see the exception/errors here? Course discussion topics are unfortunately pretty important.

UvgenGen replied:
In test `test_get_discussion_topics` `self.get_handler(self.course, handler=views.discussion_topics)` return response with empty `subcategories` and `children` for `inline_discussions`:
```
(Pdb) response['inline_discussions']
{'subcategories': {}, 'children': []}
(Pdb) response
{'course_wide_discussions': {'entries': {'Topic B': {'id': 'course_0_course_Topic B', 'sort_key': 'A', 'start_date': '2022-10-31T16:03:24.974Z', 'is_divided': True}}, 'children': [['Topic B', 'entry']]}, 'inline_discussions': {'subcategories': {}, 'children': []}}
```

ormsbee commented on lms/djangoapps/gating/tests/test_integration.py:
```diff
@@ -22,13 +23,14 @@
 from openedx.core.lib.gating import api as gating_api
 
 
+@skip("OldMongo Deprecation")
```
How much of this one fails?

UvgenGen replied:
All tests are fails by one reason - in `set_up_course` when tried create children (`cls.chapter1 = ItemFactory.create(...`)
```
lms/djangoapps/gating/tests/test_integration.py:74: in set_up_course
    cls.chapter1 = ItemFactory.create(
../venvs/edxapp/lib/python3.8/site-packages/factory/base.py:528: in create
    return cls._generate(enums.CREATE_STRATEGY, kwargs)
../venvs/edxapp/lib/python3.8/site-packages/factory/base.py:465: in _generate
    return step.build()
../venvs/edxapp/lib/python3.8/site-packages/factory/builder.py:262: in build
    instance = self.factory_meta.instantiate(
../venvs/edxapp/lib/python3.8/site-packages/factory/base.py:317: in instantiate
    return self.factory._create(model, *args, **kwargs)
xmodule/modulestore/mixed.py:81: in inner
    retval = func(field_decorator=strip_key_collection, *args, **kwargs)
xmodule/modulestore/tests/factories.py:403: in _create
    module = store.create_child(
lms/djangoapps/ccx/modulestore.py:212: in create_child
    return restore(self._modulestore.create_child(
xmodule/modulestore/mixed.py:81: in inner
    retval = func(field_decorator=strip_key_collection, *args, **kwargs)
xmodule/modulestore/mixed.py:101: in wrapper
    return func(*args, **kwargs)
xmodule/modulestore/mixed.py:763: in create_child
    return modulestore.create_child(user_id, parent_usage_key, block_type, block_id=block_id, fields=fields, **kwargs)  # lint-amnesty, pylint: disable=line-too-long
xmodule/modulestore/split_mongo/split_draft.py:191: in create_child
    self._auto_publish_no_children(item.location, item.location.block_type, user_id, **kwargs)
xmodule/modulestore/split_mongo/split_draft.py:117: in _auto_publish_no_children
    self.publish(location.version_agnostic(), user_id, blacklist=EXCLUDE_ALL, **kwargs)
xmodule/modulestore/split_mongo/split_draft.py:375: in publish
    super().copy(
xmodule/modulestore/split_mongo/split.py:2309: in copy
    self._sync_children(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <xmodule.modulestore.split_mongo.split_draft.DraftVersioningModuleStore object at 0x7f7e37b891c0>
source_parent = BlockData(fields={'display_name': 'edX 101', 'discussion_topics': {'General': {'id': 'course'}}, 'tabs': [{'type': 'co...:00, edited_by=-3, original_usage=None, original_usage_version=None, _subtree_edited_on=None, _subtree_edited_by=None))
destination_parent = BlockData(fields={'display_name': 'edX 101', 'discussion_topics': {'General': {'id': 'course'}}, 'enable_subsection_ga...0:00, edited_by=0, original_usage=None, original_usage_version=None, _subtree_edited_on=None, _subtree_edited_by=None))
new_child = BlockKey(type='chapter', id='chapter_1')

    def _sync_children(self, source_parent, destination_parent, new_child):
        """
        Reorder destination's children to the same as source's and remove any no longer in source.
        Return the removed ones as orphans (a set).
        """
        destination_reordered = []
>       destination_children = set(destination_parent.fields['children'])
E       KeyError: 'children'
```

ormsbee commented on lms/djangoapps/instructor/tests/test_enrollment.py:
```diff
@@ -368,21 +368,26 @@ def test_unenroll_norecord(self):
+            )
             cls.parent = ItemFactory(
                 category="library_content",
-                parent=cls.course,
```
Wow. I didn't think the way this was done before could even work. Never mind the weirdness of having a `library_content` module as a direct child of a CourseBlock–Old Mongo courses aren't even intended to be able to use library content. Argh. Anyhow, glad to see this change.

ormsbee commented on lms/djangoapps/instructor/tests/test_tools.py:
```diff
@@ -97,11 +97,12 @@ def test_parse_error(self):
             tools.parse_datetime('foo')
 
 
+@skip("OldMongo Deprecation")
```
Putting a reminder here for myself. I'm not sure what level of support these tools have at the moment.

ormsbee commented on lms/djangoapps/instructor_task/tests/test_integration.py:
```diff
@@ -69,6 +70,7 @@ def _assert_task_failure(self, entry_id, task_type, problem_url_name, expected_m
         assert status['message'] == expected_message
 
 
+@skip("OldMongo Deprecation")
```
These tests are likely important enough to need to fix. I need to dig into this more.

ormsbee commented on lms/djangoapps/lti_provider/management/commands/tests/test_resend_lti_scores.py:
```diff
@@ -46,15 +46,17 @@ class CommandExecutionTestCase(SharedModuleStoreTestCase):
         import_course_from_xml(
             cls.store,
-            'test_user',
+            user.id,
```
It's a little weird to me that a string got replaced with an int and this works fine, but I guess if the tests still pass. 🤷

ormsbee commented on lms/djangoapps/lti_provider/tests/test_outcomes.py:
```diff
@@ -371,69 +371,60 @@ def test_create_two_lti_consumers_with_empty_instance_guid(self):
-            assignments = outcomes.get_assignments_for_problem(
-                self.unit, self.user_id, self.course.id
-            )
+        assignments = outcomes.get_assignments_for_problem(
```
I think we could keep in the MongoDB call count checking, but I'm okay with removing it here too. Not sure it provides much value in this scenario.

ormsbee commented on lms/djangoapps/program_enrollments/rest_api/v1/tests/test_views.py:
```diff
@@ -1884,6 +1884,7 @@ def test_no_due_dates(self):
         assert status.HTTP_200_OK == response_status_code
         assert [] == response_course_runs[0]['due_dates']
 
+    @skip("OldMongo Deprecation")
```
Okay, due dates are important. We'll probably have to dig into these.

ormsbee commented on lms/djangoapps/support/tests/test_views.py:
```diff
@@ -272,6 +273,7 @@ def test_certificates_with_user_filter(self):
         response = self.client.get(url)
         self.assertContains(response, "userFilter: 'student@example.com'")
 
+    @skip("OldMongo Deprecation")
```
What breaks here?

UvgenGen replied:
problem due to escaping course id in response: `course\\u002Dv1:org.0+course_0+Run_0` instead of `course-v1:org.0+course_0+Run_0`

ormsbee commented on openedx/core/djangoapps/content/course_overviews/tests/test_course_overviews.py:
```diff
@@ -315,39 +313,34 @@ def test_get_non_existent_course(self, modulestore_type):
         assert CourseOverview.objects.filter(id=course.id).exists()
         assert CourseOverview.course_exists(course.id)
 
-    @ddt.data(ModuleStoreEnum.Type.split, ModuleStoreEnum.Type.mongo)
```
CourseOverview generation is one of the small handful of things that we will need to keep testing on both Modulestores, because CourseOverviews get regenerated whenever new attributes are added, and that process relies on being able to query the root CourseBlock of the courses in question.

UvgenGen replied:
We can't create the old mongo test course because `update_item` is deprecated and raised `NotImplementedError`

ormsbee replied:
Can we mock the data at the Mongo layer–i.e. exercise Old Mongo read logic on the root CourseBlock by mocking what it gets back from the database? CourseOverview compatibility is one of the things that has to keep working, or it will cause operational errors the next time someone adds something to CourseOverviews, which may be months from now.

UvgenGen replied:
We investigate possibility to mock draft course data, but in witch case CourseOverview can be regenerated if we removed all create/update methods for draft modulestore?

ormsbee commented on openedx/core/djangoapps/content/course_overviews/tests/test_signals.py:
```diff
@@ -54,28 +53,26 @@ def test_cache_invalidation(self, modulestore_type):
-            course.mobile_available = False
-            with self.store.branch_setting(ModuleStoreEnum.Branch.draft_preferred):
-                self.store.update_item(course, ModuleStoreEnum.UserID.test)
+        # Create a course where mobile_available is True.
```
Like the note above, CourseOverviews here need to continue testing both modulestores.

UvgenGen replied:
We can't create the old mongo test course because `update_item` is deprecated and raised `NotImplementedError`.

ormsbee commented on openedx/core/djangoapps/content/course_overviews/tests/test_tasks.py:
```diff
@@ -12,9 +11,9 @@
-        self.course_3 = CourseFactory.create(default_store=ModuleStoreEnum.Type.mongo)
+        self.course_1 = CourseFactory.create()
+        self.course_2 = CourseFactory.create()
+        self.course_3 = CourseFactory.create()
```
FWIW, this change is fine to me. CourseOverview tests that test both modulestores should stay testing both modulestores, but if there's on like this that just assumes a modulestore, we might as well switch that one over to split by default (as you do here).

ormsbee reviewed (changes requested):
Okay, there is a whole lot of stuff here, and this review is not complete. But I wanted to give some feedback reasonably soon so you could start on the next iteration of this work.

First off, congratulations. Even just getting a passing build of a 5.5K PR like this takes a lot of work to pull off.

I did not anticipate the extent to which you had to skip tests in order to get this building. I know I made the decision that it was okay to disable tests that had never worked in Split, but I didn't realize so much core functionality like access control was included in these tests, and relying on the fact that the two modulestores more were mostly compatible in behavior.

I do think that many of these tests will need to be fixed as opposed to skipped entirely, and we're going to need to triage that. This will likely increase the scope of the project some more (FYI @e0d).

So my first request is to please make a separate pull request that has all the straightforward, non-controversial test conversions. These are the test files where:

* No tests are marked "skip".
* There are no PR comments/questions about logic changes.
* You're removing Old Mongo from ddt'd tests or just changing the one modulestore being used.

You can do that before digging into any of the questions I put on this PR review. If there are borderline cases where you're unsure, don't include them in the new PR just yet. I just want to separate out the really safe test changes to reduce some of the noise here–and to merge in the simple stuff as quickly as possible before there's more drift from master.

Once that's merged and this is rebased, I want to take a look at the broad areas of what gaps in testing we're going to see–access control, forums, drafts, instructor tasks, etc.

Thank you.

zacharis278 commented on xmodule/modulestore/tests/factories.py:
```diff
@@ -366,6 +366,7 @@ def _create(cls, target_class, **kwargs):  # lint-amnesty, pylint: disable=argum
         has_score = kwargs.pop('has_score', None)
         submission_start = kwargs.pop('submission_start', None)
         submission_end = kwargs.pop('submission_end', None)
+        is_proctored_exam = kwargs.pop('is_proctored_exam', None)
```
I think this line is hiding an underlying issue (see my comment in test_exams.py). because is_proctored_exam is never actually being saved from what I can tell and causing some false positives in proctoring related tests.

UvgenGen replied:
Updated

UvgenGen commented:
@ormsbee I've opened a separate pull request with the straightforward tests:
https://github.com/openedx/edx-platform/pull/31224
This pr includes some of tests that have been commented on about `ddt`, it's fixed.

ormsbee commented:
Merged #31224. Please rebase this PR.

mphilbrick211 commented:
@UvgenGen - hi there! Just following-up on this - looks like Dave requested changes, so I'm just following up :)

UvgenGen commented:
@mphilbrick211 Hi! I'll rebased it after merging this PR https://github.com/openedx/edx-platform/pull/31480

mphilbrick211 commented:
Flagging this @ormsbee!

UvgenGen committed:
feat!: Remove support for children in Old Mongo

UvgenGen commented:
@ormsbee @mphilbrick211 PR rebased

mphilbrick211 commented:
Thanks @UvgenGen! @ormsbee flagging this for you, though I know you are traveling this week.

mphilbrick211 commented:
> Thanks @UvgenGen! @ormsbee flagging this for you, though I know you are traveling this week.

Friendly ping on this @ormsbee :)

ormsbee commented:
@mphilbrick211: Thank you for the reminder. FWIW, I did start re-reading through this today, but it's taking me a while. Aiming to have a review submitted by end of day Wed.

ormsbee commented on lms/djangoapps/course_api/tests/test_serializers.py:
```diff
@@ -40,45 +39,41 @@ def setUp(self):
             'number': 'toy',
             'org': 'edX',
-            'short_description': 'A course about toys.',
+            'short_description': None,
```
Is there something that breaks when the `short_description` is filled in?

UvgenGen replied:
For ToyCourseFactory `short_description` and some other course items are created for the `draft_preferred` branch.
https://github.com/raccoongang/edx-platform/blob/sagirov/EDXOLDMNG-146/xmodule/modulestore/tests/factories.py#L224
I'll update MODULESTORE setting for this tests.

ormsbee commented on xmodule/modulestore/mongo/base.py:
```diff
@@ -92,50 +85,38 @@ class MongoKeyValueStore(InheritanceKeyValueStore):
 
     def get(self, key):
-        if key.scope == Scope.children:
-            return self._children
```
This looks like we're raising an error when child access is attempted. Can we make this always return an empty list instead? So naive code that just wants to iterate through children sees an empty course instead of throwing an error?

ormsbee reviewed (changes requested):
A comment, a small comment request, and a few compatibility related requests/questions.

Sorry for the long delay. This can be merged next week once the compatibility issues are addressed, but should be coordinated with @jristau1984 since there is a risk of breaking things. I think there's relatively low risk of breaking Studio or LMS directly, but a higher risk of breaking automated tasks that might read course data for various things (e.g. course data package exports).

UvgenGen committed:
feat: code review

UvgenGen committed:
fix: fixed PluginMissingError for split modulestore

ormsbee commented:
@UvgenGen: Is this ready for another review? (I just want to make sure you're not waiting on me.)

UvgenGen commented:
@ormsbee I addressed all comments except last one. I left a question to your comment: 
> Can we mock the data at the Mongo layer–i.e. exercise Old Mongo read logic on the root CourseBlock by mocking what it gets back from the database? CourseOverview compatibility is one of the things that has to keep working, or it will cause operational errors the next time someone adds something to CourseOverviews, which may be months from now.

Q:
> We investigate possibility to mock draft course data, but in witch case CourseOverview can be regenerated if we removed all create/update methods for draft modulestore?

PR updated except this issue. I will wait for an answer, thanks)

ormsbee commented:
> We investigate possibility to mock draft course data, but in witch case CourseOverview can be regenerated if we removed all create/update methods for draft modulestore?

I'm not entirely sure if I understand this question, so I'll reiterate my priorities:

1. It is *required* that we still be able to regenerate CourseOverviews from Old Mongo courses, which requires being able to read the root CourseBlock from those courses. Breaking this could potentially break a lot of other functionality that we don't want to deal with yet.
2. If we can ensure that CourseOverviews will continue to work by mocking the data that comes back from MongoDB in situations where Old Mongo CourseOverviews are generated, then let's do that. If this works, we can remove the create/update methods on the old modulestore.

Does that make sense?

UvgenGen commented:
@ormsbee we are not found any test that regenerate course overview, so we planning add a new one. If it's possible to regenerate course overview we will follow you advice.

UvgenGen committed:
test: added test for mongo course overview generation

UvgenGen commented:
@ormsbee we added new course overview test for Old Mongo:
`openedx/core/djangoapps/content/course_overviews/tests/test_course_overviews.py::CourseOverviewTestCase::test_mongo_course_overview_generation`
Please take a look)

ormsbee reviewed (approved):
Thanks folks. Now we just have to work out the merge timing.

FYI @jristau1984

mphilbrick211 commented:
Hi @UvgenGen @ormsbee! Looks like tests need to be run on this PR. Is there an update on getting this merged?

UvgenGen committed:
fix: fix linter error

UvgenGen committed:
test: update video/lti block tests

ormsbee commented:
@mphilbrick211: Just wanted to merge the last of the BD-13 PRs before this one.

@kaustavb12, @Agrendalath: @glugovgrglib mentioned that [new failures](https://github.com/openedx/edx-platform/actions/runs/4811655665/jobs/8565912731?pr=31134#step:7:98) occurred on this PR after https://github.com/openedx/edx-platform/pull/31472 was merged. Do you have any insight on this?

@jristau1984: Assuming that the above issue gets settled, would it be okay to merge this on Monday?

Agrendalath commented:
@ormsbee, https://github.com/openedx/edx-platform/pull/31472 should not cause `SyntaxError: invalid syntax`  errors. I don't have any context on this PR, but it seems that a force push + 581f07c47d0644b446c3e7751b7100992fbafb2c resolved these failures?

ormsbee commented:
@Agrendalath: I think it's more a question of whether that https://github.com/openedx/edx-platform/commit/581f07c47d0644b446c3e7751b7100992fbafb2c should have been necessary–i.e. why did the calls to `block.render[STUDENT_VIEW]` break after #31472 merged?

Agrendalath commented:
@ormsbee, I ran `lms/djangoapps/courseware/tests/test_video_mongo.py::TestVideoNonYouTube::test_video_constructor` and checked the generated `context`. It looks like this.

`master` branch before https://github.com/openedx/edx-platform/pull/31472:
```python
(('video.html',
  {'autoadvance_enabled': False,
   'branding_info': None,
   'bumper_metadata': 'null',
   'cdn_eval': False,
   'cdn_exp_group': None,
   'display_name': 'A Name',
   'download_video_link': 'example.mp4',
   'handout': None,
   'hide_downloads': False,
   'id': 'i4x-org_0-course_0-video-A_Name',
   'is_embed': False,
   'license': None,
   'metadata': '{"autoAdvance": false, "autohideHtml5": false, "autoplay": '
               'false, "captionDataDir": null, "completionEnabled": false, '
               '"completionPercentage": 0.95, "duration": null, "end": 3610.0, '
               '"generalSpeed": 1.0, "lmsRootURL": "http://localhost:8000", '
               '"poster": null, "prioritizeHls": false, '
               '"publishCompletionUrl": '
               '"i4x://org.0/course_0/video/A_Name/publish_completion", '
               '"recordedYoutubeIsAvailable": true, "savedVideoPosition": 0.0, '
               '"saveStateEnabled": true, "saveStateUrl": '
               '"i4x://org.0/course_0/video/A_Name/xmodule_handler/save_user_state", '
               '"showCaptions": "true", "sources": ["example.mp4", '
               '"example.webm"], "speed": null, "start": 3603.0, "streams": '
               '"1.00:3_yD_cEKoCk", "transcriptAvailableTranslationsUrl": '
               '"i4x://org.0/course_0/video/A_Name/transcriptavailable_translations", '
               '"transcriptLanguage": "en", "transcriptLanguages": {"en": '
               '"English"}, "transcriptTranslationUrl": '
               '"i4x://org.0/course_0/video/A_Name/transcripttranslation/__lang__", '
               '"ytApiUrl": "https://www.youtube.com/iframe_api", '
               '"ytMetadataEndpoint": "", "ytTestTimeout": 1500}',
   'poster': 'null',
   'public_video_url': None,
   'track': None,
   'transcript_download_format': 'srt',
   'transcript_download_formats_list': [{'display_name': 'SubRip (.srt) file',
                                         'value': 'srt'},
                                        {'display_name': 'Text (.txt) file',
                                         'value': 'txt'}]}),
 {})
```

`master` branch:
```python
(('video.html',
  {'autoadvance_enabled': False,
   'branding_info': None,
   'bumper_metadata': 'null',
   'cdn_eval': False,
   'cdn_exp_group': None,
   'display_name': 'A Name',
   'download_video_link': 'example.mp4',
   'handout': None,
   'hide_downloads': False,
   'id': 'i4x-org_0-course_0-video-A_Name',
   'is_embed': False,
   'license': None,
   'metadata': '{"autoAdvance": false, "autohideHtml5": false, "autoplay": '
               'false, "captionDataDir": null, "completionEnabled": false, '
               '"completionPercentage": 0.95, "duration": null, "end": 3610.0, '
               '"generalSpeed": 1.0, "lmsRootURL": "http://localhost:8000", '
               '"poster": null, "prioritizeHls": false, '
               '"publishCompletionUrl": '
               '"/courses/org.0/course_0/Run_0/xblock/i4x:;_;_org.0;_course_0;_video;_A_Name/handler/publish_completion", '
               '"recordedYoutubeIsAvailable": true, "savedVideoPosition": 0.0, '
               '"saveStateEnabled": true, "saveStateUrl": '
               '"/courses/org.0/course_0/Run_0/xblock/i4x:;_;_org.0;_course_0;_video;_A_Name/handler/xmodule_handler/save_user_state", '
               '"showCaptions": "true", "sources": ["example.mp4", '
               '"example.webm"], "speed": null, "start": 3603.0, "streams": '
               '"1.00:3_yD_cEKoCk", "transcriptAvailableTranslationsUrl": '
               '"/courses/org.0/course_0/Run_0/xblock/i4x:;_;_org.0;_course_0;_video;_A_Name/handler/transcript/available_translations", '
               '"transcriptLanguage": "en", "transcriptLanguages": {"en": '
               '"English"}, "transcriptTranslationUrl": '
               '"/courses/org.0/course_0/Run_0/xblock/i4x:;_;_org.0;_course_0;_video;_A_Name/handler/transcript/translation/__lang__", '
               '"ytApiUrl": "https://www.youtube.com/iframe_api", '
               '"ytMetadataEndpoint": "", "ytTestTimeout": 1500}',
   'poster': 'null',
   'public_video_url': None,
   'track': None,
   'transcript_download_format': 'srt',
   'transcript_download_formats_list': [{'display_name': 'SubRip (.srt) file',
                                         'value': 'srt'},
                                        {'display_name': 'Text (.txt) file',
                                         'value': 'txt'}]}),
 {})
```

Current version (from this PR):
```python
(('video.html',
  {'autoadvance_enabled': False,
   'branding_info': None,
   'bumper_metadata': 'null',
   'cdn_eval': False,
   'cdn_exp_group': None,
   'display_name': 'A Name',
   'download_video_link': 'example.mp4',
   'handout': None,
   'hide_downloads': False,
   'id': 'A_Name',
   'is_embed': False,
   'license': None,
   'metadata': '{"autoAdvance": false, "autohideHtml5": false, "autoplay": '
               'false, "captionDataDir": null, "completionEnabled": false, '
               '"completionPercentage": 0.95, "duration": null, "end": 3610.0, '
               '"generalSpeed": 1.0, "lmsRootURL": "http://localhost:8000", '
               '"poster": null, "prioritizeHls": false, '
               '"publishCompletionUrl": '
               '"/courses/course-v1:org.0+course_0+Run_0/xblock/block-v1:org.0+course_0+Run_0+type@video+block@A_Name/handler/publish_completion", '
               '"recordedYoutubeIsAvailable": true, "savedVideoPosition": 0.0, '
               '"saveStateEnabled": true, "saveStateUrl": '
               '"/courses/course-v1:org.0+course_0+Run_0/xblock/block-v1:org.0+course_0+Run_0+type@video+block@A_Name/handler/xmodule_handler/save_user_state", '
               '"showCaptions": "true", "sources": ["example.mp4", '
               '"example.webm"], "speed": null, "start": 3603.0, "streams": '
               '"1.00:3_yD_cEKoCk", "transcriptAvailableTranslationsUrl": '
               '"/courses/course-v1:org.0+course_0+Run_0/xblock/block-v1:org.0+course_0+Run_0+type@video+block@A_Name/handler/transcript/available_translations", '
               '"transcriptLanguage": "en", "transcriptLanguages": {"en": '
               '"English"}, "transcriptTranslationUrl": '
               '"/courses/course-v1:org.0+course_0+Run_0/xblock/block-v1:org.0+course_0+Run_0+type@video+block@A_Name/handler/transcript/translation/__lang__", '
               '"ytApiUrl": "https://www.youtube.com/iframe_api", '
               '"ytMetadataEndpoint": "", "ytTestTimeout": 1500}',
   'poster': 'null',
   'public_video_url': None,
   'track': None,
   'transcript_download_format': 'srt',
   'transcript_download_formats_list': [{'display_name': 'SubRip (.srt) file',
                                         'value': 'srt'},
                                        {'display_name': 'Text (.txt) file',
                                         'value': 'txt'}]}),
 {})
```

Version from before https://github.com/openedx/edx-platform/commit/581f07c47d0644b446c3e7751b7100992fbafb2c:
```python
(('video.html',
  {'autoadvance_enabled': False,
   'branding_info': None,
   'bumper_metadata': 'null',
   'cdn_eval': False,
   'cdn_exp_group': None,
   'display_name': 'A Name',
   'download_video_link': 'example.mp4',
   'handout': None,
   'hide_downloads': False,
   'id': 'A_Name',
   'is_embed': False,
   'license': None,
   'metadata': '{"autoAdvance": false, "autohideHtml5": false, "autoplay": '
               'false, "captionDataDir": null, "completionEnabled": false, '
               '"completionPercentage": 0.95, "duration": null, "end": 3610.0, '
               '"generalSpeed": 1.0, "lmsRootURL": "http://localhost:8000", '
               '"poster": null, "prioritizeHls": false, '
               '"publishCompletionUrl": '
               '"/courses/course-v1:org.0+course_0+Run_0/xblock/block-v1:org.0+course_0+Run_0+type@video+block@A_Name/handler/publish_completion", '
               '"recordedYoutubeIsAvailable": true, "savedVideoPosition": 0.0, '
               '"saveStateEnabled": true, "saveStateUrl": '
               '"/courses/course-v1:org.0+course_0+Run_0/xblock/block-v1:org.0+course_0+Run_0+type@video+block@A_Name/handler/xmodule_handler/save_user_state", '
               '"showCaptions": "true", "sources": ["example.mp4", '
               '"example.webm"], "speed": null, "start": 3603.0, "streams": '
               '"1.00:3_yD_cEKoCk", "transcriptAvailableTranslationsUrl": '
               '"/courses/course-v1:org.0+course_0+Run_0/xblock/block-v1:org.0+course_0+Run_0+type@video+block@A_Name/handler/transcript/available_translations", '
               '"transcriptLanguage": "en", "transcriptLanguages": {"en": '
               '"English"}, "transcriptTranslationUrl": '
               '"/courses/course-v1:org.0+course_0+Run_0/xblock/block-v1:org.0+course_0+Run_0+type@video+block@A_Name/handler/transcript/translation/__lang__", '
               '"ytApiUrl": "https://www.youtube.com/iframe_api", '
               '"ytMetadataEndpoint": "", "ytTestTimeout": 1500}',
   'poster': 'null',
   'public_video_url': None,
   'track': None,
   'transcript_download_format': 'srt',
   'transcript_download_formats_list': [{'display_name': 'SubRip (.srt) file',
                                         'value': 'srt'},
                                        {'display_name': 'Text (.txt) file',
                                         'value': 'txt'}]}),
 {})

<div class="aside-block"
    data-success-class="fa fa-check-square-o fa-lg pass"
    data-failure-class="fa fa-times fa-lg fail"
    data-error-class="fa fa-exclamation-triangle fa-lg error"
    data-unknown-class="fa fa-question-circle fa-lg unknown"
    data-local-resource-url="/static/xblock/resources/acid/public/test_data.json"
>
    <button>Acid Aside for block-v1:org.0+course_0+Run_0+type@video+block@A_Name</button>
    <div>
        <p>JS init function run:
            <span class="js-init-run">
                <i class="fa fa-question-circle fa-lg unknown"></i>
            </span>
        </p>
        <p>Resource Url Test:
            <span class="local-resource-test">
                <i class="fa fa-question-circle fa-lg unknown"></i>
            </span>
        </p>
        <table class='storage-tests'>
            <tr>
                <th>Scope</th>
                <th>Server-side<br>handler_url<br>returned</th>
                <th>Server-side<br>handler_url<br>succeeded</th>
                <th>Client-side<br>handler_url<br>returned</th>
                <th>Client-side<br>handler_url<br>succeeded</th>
            </tr>
                <tr class="scope-storage-test scope-user_state "
                    data-handler-url="/courses/course-v1:org.0+course_0+Run_0/xblock/aside-usage-v2:block-v1$:org.0+course_0+Run_0+type@video+block@A_Name::acid_aside/handler/check_storage/SUFFIX1465?QUERY=1465&SCOPE=user_state"
                    data-scope="user_state"
                    data-value="1465"
                >
                    <td>user_state</td>
                    <td>
                        <span class="server-storage-test-returned">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="server-storage-test-succeeded">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="client-storage-test-returned">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="client-storage-test-succeeded">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                </tr>
                <tr class="scope-storage-test scope-preferences alt"
                    data-handler-url="/courses/course-v1:org.0+course_0+Run_0/xblock/aside-usage-v2:block-v1$:org.0+course_0+Run_0+type@video+block@A_Name::acid_aside/handler/check_storage/SUFFIX7096?QUERY=7096&SCOPE=preferences"
                    data-scope="preferences"
                    data-value="7096"
                >
                    <td>preferences</td>
                    <td>
                        <span class="server-storage-test-returned">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="server-storage-test-succeeded">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="client-storage-test-returned">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="client-storage-test-succeeded">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                </tr>
                <tr class="scope-storage-test scope-user_info "
                    data-handler-url="/courses/course-v1:org.0+course_0+Run_0/xblock/aside-usage-v2:block-v1$:org.0+course_0+Run_0+type@video+block@A_Name::acid_aside/handler/check_storage/SUFFIX4376?QUERY=4376&SCOPE=user_info"
                    data-scope="user_info"
                    data-value="4376"
                >
                    <td>user_info</td>
                    <td>
                        <span class="server-storage-test-returned">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="server-storage-test-succeeded">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="client-storage-test-returned">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="client-storage-test-succeeded">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                </tr>
                <tr class="scope-storage-test scope-user_state_summary alt"
                    data-handler-url="/courses/course-v1:org.0+course_0+Run_0/xblock/aside-usage-v2:block-v1$:org.0+course_0+Run_0+type@video+block@A_Name::acid_aside/handler/check_storage/SUFFIX2238?QUERY=2238&SCOPE=user_state_summary"
                    data-scope="user_state_summary"
                    data-value="2238"
                >
                    <td>user_state_summary</td>
                    <td>
                        <span class="server-storage-test-returned">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="server-storage-test-succeeded">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="client-storage-test-returned">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                    <td>
                        <span class="client-storage-test-succeeded">
                            <i class="fa fa-question-circle fa-lg unknown"></i>
                        </span>
                    </td>
                </tr>
        </table>
        <p>Sees Acid DOM: <span class="acid-dom"><i class="fa fa-question-circle fa-lg unknown"></i></span></p>
    </div>
</div>
```

It seems that https://github.com/openedx/edx-platform/pull/31472 changed the URLs in this context, but it didn't trigger any test failures. Therefore, the behavior change of `self.block.render(STUDENT_VIEW)` might be caused by this PR (https://github.com/openedx/edx-platform/pull/31134).

jristau1984 commented:
@ormsbee there are several production issues in-flight right now, I will have to wait to merge this until the impact from those issues dies down a bit.

UvgenGen commented:
@ormsbee PR is rebased but I have an error with JS tests. Looks like this pipeline needs to be restarted

ormsbee commented on lms/djangoapps/courseware/tests/test_lti_integration.py:
```diff
@@ -117,7 +116,7 @@ def mocked_sign(self, *args, **kwargs):
 
     def test_lti_constructor(self):
-        generated_content = self.block.render(STUDENT_VIEW).content
+        generated_content = self.block.student_view(None).content
```
This is the one thing that's holding up this PR for me at this point. Why does `self.block.render(STUDENT_VIEW).content` no longer work?

UvgenGen replied:
Student view content has been updated (html was added in content), it is fully described here: https://github.com/openedx/edx-platform/pull/31134#issuecomment-1527658373
That's why I use a similar method

ormsbee replied:
Okay, I realize I'm probably being very dense here, but why was html added in the content?

My basic concern is that we've somehow changed the behavior of the `render` method, which doesn't seem like it should be necessary when removing Old Mongo functionality. If this PR does change the `render` method behavior, why does it do that? If this PR does not change the `render` method behavior, why is this change to the test necessary?

I get that HTML was added to the content, but I don't understand _why_ HTML was added to the content.

Thank you.

ormsbee replied:
@UvgenGen: Do you have time to chat about this in the next couple of days? I think maybe I'm just not understanding it properly. Please ping me on Slack when you have a moment.

UvgenGen committed:
fix: update TestMongoKeyValueStore tests

openedx-webhooks commented:
@UvgenGen 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/31134&entry.1484655318=https://github.com/UvgenGen&entry.752974735=2022-10-10+09:41&entry.1917517419=2023-09-06+14:01&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
